docs/architecture/adr-019-inflation-rate.md

```diff
@@ -0,0 +1,112 @@
+# ADR 019: Inflation rate
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/3/31: initial draft
+
+## Context
+
+The inflation rate is decided to have a fixed rate with an initial value and a target value.
+Table below shows that how it looks like according to the made decision.
+
+| Name                       | Value  |
+|-------------------------------|--------|
+| Initial Inflation          | 8.00%  |
+| Disinflation rate per year    | 10.00% |
+| Target Inflation (floor)   | 1.50%  |
+
+Once we approach the target inflation, we stick to it forever.
+The table below depicts the values over the coming years:
+
+| year | inflation (%) |
+|------|------|
+|0  | 8.00 |
+|1  | 7.20 |
+|2  | 6.48 |
+|3  | 5.832 |
+|4  | 5.2488 |
+|5  | 4.72392 |
+|6  | 4.251528 |
+|7  | 3.8263752 |
+|8  | 3.44373768 |
+|9  | 3.099363912 |
+|10 | 2.7894275208 |
+|11 | 2.51048476872 |
+|12 | 2.259436291848 |
+|13 | 2.0334926626632 |
+|14 | 1.83014339639688 |
+|15 | 1.647129056757192 |
+|16 | 1.50 |
+|17 | 1.50 |
+|18 | 1.50 |
+|19 | 1.50 |
+|20 | 1.50 |
+
+## Decision
+
+The decision is already made and the ADR is accepted. We are waiting for a final inspection though.
+
+## Detailed Design
+
+In order to do so, we need to have a simple mint module. So we remove the dynamic inflation rate calculation and simply map the current block height to the respective year and calculate the inflation rate according to that.
+
+The logic is implemented in the `NextInflationRate` function and is called in the `BeginBlock` of the `mint` module.
+
+If no custom inflation calculator function is passed to the Minter, the default function calculates the inflation rate that this ADR desires to achieve.
+
+Since the rates are fixed, we hardcoded them in a set of constants.
+
+```go
+const (
+	InitialInflationRate    = 0.08
+	TargetInflationRate     = 0.015 // floor
+	DisinflationRatePerYear = 0.1
+)
+```
+
+The `NextInflationRate` function calculates the `year` based on the current block height that comes from the `sdk.Context` and the number of `BlocksPerYear` that comes from the params.
+Then it computes the inflation rate according to the determined fixed rate per year.
+
+```go
+NextInflationRate(ctx sdk.Context, params Params) sdk.Dec {
+
+	year := uint64(ctx.BlockHeader().Height) / params.BlocksPerYear
+
+	initInflationRate := sdk.NewDecWithPrec(InitialInflationRate*1000, 3 /* since we used 1000 */)
+	targetInflationRate := sdk.NewDecWithPrec(TargetInflationRate*1000, 3 /* since we used 1000 */)
+
+	// initInflationRate * ((1 - DisinflationRate) ^ year)
+	newInflationRate := initInflationRate.Mul(
+		sdk.OneDec().Sub(
+			sdk.NewDecWithPrec(DisinflationRatePerYear*100, 2 /* since we used 100 */)).
+			Power(year))
+
+	if newInflationRate.LT(targetInflationRate) {
+		newInflationRate = targetInflationRate
+	} else {
+		if newInflationRate.GT(initInflationRate) {
+			newInflationRate = initInflationRate
+		}
+	}
+
+	return newInflationRate
+}
```

Comment:
if it is time based would it be deterministic? what level of clock drift does tendermint tolerate?

Reply:
It would be deterministic if the time in the header is used, which is determined by [the median](https://github.com/celestiaorg/celestia-core/blob/ed981bbc97f8068b02b7fc52a05c4ae550b8cc20/state/state.go#L244-L250) of each time provided by the validators in the last commit.

> what level of clock drift does tendermint tolerate?

I could be missing something, but it looks like the only [check](https://github.com/celestiaorg/celestia-core/blob/ed981bbc97f8068b02b7fc52a05c4ae550b8cc20/state/validation.go#L116-L129) is that the time is increasing, and that it actually matches the median time.

Reply:
It might be simpler to reason about a block-based instead of time-based schedule. Then it should be communicated as that: every X blocks (which roughly corresponds to one year) the disinflation param will be applied. 

That said, I still wonder if the current flow could not be simplified. Roughly to: if a year has passed (in blocks/or via timestamp), calculate and return next inflation, else simply return current inflation.

Reply:
I do think time based might be better, because block interval is not a consensus critical parameter, so block producers could print more money by reducing their block interval.

It would also be possible for validators to manipulate the time, but I think would be harder given it's based on median time. Plus if they change the time to the future that would be non-ambitiously malicious. Whereas if a small minority of validators reduce their block interval, that could have an impact on emissions.

Reply:
I also agree that time seems to be the best approach here. It seems like it could help simplify the approach because we don't have to think about historic expected blocks per year, and also this is what osmosis does to schedule/[trigger epochs.](https://github.com/osmosis-labs/osmosis/blob/087a97b2976b989f94ab52000a64377c0bcf1a79/x/epochs/keeper/abci.go#L19-L22)

> It would also be possible for validators to manipulate the time, but I think would be harder given it's based on median time. Plus if they change the time to the future that would be non-ambitiously malicious. 

+1, after digging a bit deeper, at least offchain [light client verification](https://github.com/celestiaorg/celestia-core/blob/ed981bbc97f8068b02b7fc52a05c4ae550b8cc20/light/verifier.go#L176-L181) would fail